### PR TITLE
Pre commit fixes

### PR DESCRIPTION
**Pull request by m-iwanicki**

### CODE REVIEW

**Comment by m-iwanicki** [comment]
lychee failures:

```
[docs/community.md]:
✗ [500] https://slack.osfw.dev/ | Failed: Network error: Internal Server Error

[docs/events.md]:
✗ [404] https://www.youtube.com/watch?v=A9GrlQsQc7Q?start=8152 | Failed: Network error: Not Found
✗ [429] https://vimeo.com/488140434 | Failed: Network error: Too Many Requests
```

Both vimeo and youtube work so only problem is `slack.osfw.dev`

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    hooks:
+      - id: codespell
+
+  - repo: https://github.com/adrienverge/yamllint
+    rev: v1.35.1
+    hooks:
+      - id: yamllint
+        args: [-c=.yamllint]
+
+  - repo: https://github.com/3mdeb/hooks
```
Comment (krystian-hebel) [inline .pre-commit-config.yaml]: I think this isn't necessary, this projects doesn't overlap with our product names.

Comment (m-iwanicki) [reply .pre-commit-config.yaml]: This hook checks more than just our product names. Currently it also checks Yocto/UEFI/NVIDIA and we will probably add more names/trademarks in the future.

**Comment by m-iwanicki** [comment]
In addition to previous lychee failures there is also:

* https://mirroronet.pl/pub/mirrors/FOSDEM/2020/K.4.601/firmware_itsoecs.mp4 - video can't be played becase the file is corrupt
* https://www.amd.com/system/files/TechDocs/24593.pdf - redirected, doesn't go directly to wanted document

File: .pre-commit-config.yaml (hunk 1 above)
Comment (krystian-hebel) [reply .pre-commit-config.yaml]: Is this check case-sensitive and does it skip URLs? If yes, I can approve adding it _after_ `TrenchBoot` is added to the list of names, otherwise I don't see how it helps.

File: LICENSE.md [hunk 2]
```diff
@@ -1,159 +1,334 @@
...
 
-a. ___Attribution.___
+1. If You Share the Licensed Material (including in modified form), You must:
 
-   1. If You Share the Licensed Material (including in modified form), You must:
+   A. retain the following if it is supplied by the Licensor with the Licensed
+   Material:
 
-       A. retain the following if it is supplied by the Licensor with the Licensed Material:
+   i. identification of the creator(s) of the Licensed Material and any others
```
Comment (krystian-hebel) [inline LICENSE.md]: This and following points with roman numbering aren't properly indented.

File: LICENSE.md [hunk 3]
```diff
@@ -1,159 +1,334 @@
...
+> [creativecommons.org/policies](http://creativecommons.org/policies), Creative
+> Commons does not authorize the use of the trademark “Creative Commons” or any
+> other trademark or logo of Creative Commons without its prior written consent
+> including, without limitation, in connection with any unauthorized
+> modifications to any of its public licenses or any other arrangements,
+> understandings, or agreements concerning use of licensed material. For the
+> avoidance of doubt, this paragraph does not form part of the public licenses.
 >
-> Creative Commons may be contacted at creativecommons.org
+> > Creative Commons may be contacted at creativecommons.org
```
Comment (krystian-hebel) [inline LICENSE.md]: Remove the second `>`.

File: docs/CNAME [hunk 4]
```diff
@@ -1 +1 @@
-trenchboot.org
```
Comment (krystian-hebel) [inline docs/CNAME]: This isn't documented, but supposedly it won't work if there is newline character in the file. Please revert this change and add it as an exception, so it won't cause any delayed, hard to debug DNS problems.

File: docs/FAQ.md [hunk 5]
```diff
@@ -1,51 +1,51 @@
...
-For Linux systems doing both verified(secure) and measured boot, there is an
-intermediary that handles the security enforcement. For verified boot it is the
-UEFI shim loader and for measured boot it is tboot. TrenchBoot replaces these
-intermediary loaders with a common Linux-based loader that provides a rich
-security processing framework. One role that TrenchBoot does not fulfill is
-that the UEFI shim also serves as a trust delegation point that transitions
-from Microsoft Authority to Distribution/Installer/No Authority. The response
-why this is not of concern will be addressed in Question 2.
-
+For Linux systems doing both verified(secure) and measured boot, there
```
Comment (krystian-hebel) [inline docs/FAQ.md]:
```suggestion
For Linux systems doing both verified (secure) and measured boot, there
```

File: docs/blueprints/Measured_Secure_Boot.md [hunk 6]
```diff
@@ -1,61 +1,65 @@
...
+MSB will consist of an enhancement to the GNU GRUB bootloader (grub),
+an extended Linux kernel, and the uroot initramfs environment. The
+enhancement to grub will be to add TPM 1.2/2.0 support along with
+relocators for AMD and Intel Late Launch instructions. The Linux
+kernel will be extend to function as a post-launch kernel that will
+run the TSE. Details for each of these are documented in their own
+respective blueprints. Finally, below is the execution flow for MSB
+for comparison with tboot's execution flow above.
+
+```shell
```
Comment (krystian-hebel) [inline docs/blueprints/Measured_Secure_Boot.md]:
```suggestion
```text
```

File: docs/community.md [hunk 7]
```diff
@@ -1,6 +1,5 @@
+# How to get in contact with the team
 
-How to get in contact with the team.
-
-* [TrenchBoot Google Group](https://groups.google.com/g/trenchboot-devel)
-* [Twitter](https://twitter.com/TrenchBoot)
-* [#trenchboot channel on OSFW Slack](https://slack.osfw.dev)
+- [TrenchBoot Google Group](https://groups.google.com/g/trenchboot-devel)
+- [Twitter](https://twitter.com/TrenchBoot)
+- [#trenchboot channel on OSFW Slack](https://slack.osfw.dev)
```
Comment (krystian-hebel) [inline docs/community.md]:
```suggestion
- [TrenchBoot room on Matrix](https://matrix.to/#/#OSFW-Trenchboot:matrix.org)
```

Slack channel is pretty much dead.

File: docs/steering-committee/.pages.yml [hunk 8]
```diff
@@ -1,2 +1,3 @@
+---
```
Comment (krystian-hebel) [inline docs/steering-committee/.pages.yml]: Please remove this file instead, it isn't used (I think).

File: docs/steering-committee/index.md [hunk 9]
```diff
@@ -1,6 +1,6 @@
-# Steering Commitee Meetings
+# Steering Committee Meetings
 
-Here you will find the meeting minutes from the Steering Committiee and
-Community Meetings.
+Here you will find the meeting minutes from the Steering Committiee
```
Comment (krystian-hebel) [inline docs/steering-committee/index.md]:
```suggestion
Here you will find the meeting minutes from the Steering Committee
```

Are there any spellcheckers that can be added to the CI?

File: docs/user-docs/test_matrix.md [hunk 10]
```diff
@@ -34,23 +36,13 @@ Devices and configurations on which TrenchBoot is known to not work:
 |:---------------------------------------------------------------------------------------------:|:----------:|:----------:|:-----:|
 | Supermicro M11SDV-8CT<br>(AMD EPYC 3000 Snowy Owl server)<br>(v0.4 by openQA)                 |  TPM 2.0   | 2019-today | CSM legacy boot,<br>unexpected PCR values |
 
-[aem]: https://github.com/TrenchBoot/qubes-antievilmaid
```
Comment (krystian-hebel) [inline docs/user-docs/test_matrix.md]: I think that keeping those close to where they are referenced make it easier to work with (modify, add new entries, move a part of document elsewhere etc.). Would it be possible to disable whatever check suggested this change?

File: mkdocs.yml [hunk 11]
```diff
@@ -10,34 +11,34 @@ edit_uri: ""
 copyright: Copyright &copy; 2018 - 2024 TrenchBoot Developers
 
 nav:
-    - Home: index.md
-    - Events:
+  - Home: index.md
```
Comment (krystian-hebel) [inline mkdocs.yml:14]: First indent level is 2 spaces, and all of the following ones are 4, is this what lint suggested?

File: .markdownlint.yaml [hunk 12]
```diff
@@ -0,0 +1,196 @@
...
+
+# MD012/no-multiple-blanks - Multiple consecutive blank lines
+MD012:
+  # Consecutive blank lines
+  maximum: 1
+# MD013/line-length - Line length
+#
+MD013:
+  # Number of characters
+  line_length: 80
```
Comment (krystian-hebel) [inline .markdownlint.yaml:68]: All of the files were wrapped to lower number of characters per line, why is that?

Comment (m-iwanicki) [reply .markdownlint.yaml:68]: From what I can see markdownlint-fix doesn't wrap lines automatically so it's possible it was done manually or with other tool that was configured differently.

File: mkdocs.yml (hunk 11 above)
Comment (m-iwanicki) [reply mkdocs.yml:14]: I think it depends on config: https://yamllint.readthedocs.io/en/stable/rules.html#module-yamllint.rules.indentation

Currently from what I can see it wants 2 spaces between first character in word (that is not including `-`) in current line and first character (including `-`) in next line. In examples below I used dots for those 2 spaces:


*  ```yaml
    text:
    ..- text2
    ```

*  ```yaml
    - text:
      ..- text2
    ```

*  ```yaml
    text:
    ..text2
    ```

*  ```yaml
    - text:
      ..text2
    ```

File: docs/user-docs/test_matrix.md (hunk 10 above)
Comment (m-iwanicki) [reply docs/user-docs/test_matrix.md]: From what I can see currently nothing complains about that, so I'm not sure if @EduKav1813 wasn't overeager or maybe it was victim of another auto fix.

Changed it back here: 9611f9774549a0cd30d4c2338c5017f8dfef3c32

File: docs/steering-committee/index.md (hunk 9 above)
Comment (m-iwanicki) [reply docs/steering-committee/index.md]: Currently there is one, `codespell`, from what I can see it only works by looking for explicitly defined misspelled words (common misspellings). It suggested `Commitee` -> `Committee` change in the same file.
If you want to add/change to different spellchecker then maybe create new issue here or in https://github.com/3mdeb/hooks so we could decide on best one that can be used everywhere.

Comment (m-iwanicki) [reply docs/steering-committee/index.md]: fixed in 41cb961b084915a712daf5701954265d02f7bb14

File: .pre-commit-config.yaml (hunk 1 above)
Comment (m-iwanicki) [reply .pre-commit-config.yaml]: Ok removed it. This hook only checked if defined names/trademarks are correctly capitalized (e.g. UEFI which is used in this repo)
Maybe we will add it later when it's more mature.
47d18c03eba24459d5ad1e2ed1148bb5b9e26f8a

File: LICENSE.md (hunk 2 above)
Comment (m-iwanicki) [reply LICENSE.md]: ~~From what I can see they also weren't earlier~~. Nevermind, they were indented but it doesn't render correctly.
Markdown doesn't support roman numerals by default, I think it really depends how/where it's viewed. On GitHub at least in comments you can:

```
1. a
  1. a
  1. a
  1. a
1. a
```

which will result in:

1. a
    1. a
    1. a
    1. a
1. a

or add css in license document with custom styling.

```markdown
<style type="text/css">
    ol ol { list-style-type: upper-alpha; }
    ol ol ol { list-style-type: lower-roman; }
</style>
```

But still we would need to use automatic ordering (each list point would be `1.` in raw text)

Comment (m-iwanicki) [reply LICENSE.md]: I added indent in 453b9ef336495333f86fe3fa5215a6b6595b7190 but it doesn't show when viewed on github.

File: docs/FAQ.md (hunk 5 above)
Comment (m-iwanicki) [reply docs/FAQ.md]: fixed 38411e46ed59e835bd55a8351bc3755265a420b3

File: docs/blueprints/Measured_Secure_Boot.md (hunk 6 above)
Comment (m-iwanicki) [reply docs/blueprints/Measured_Secure_Boot.md]: fixed 38411e46ed59e835bd55a8351bc3755265a420b3

File: docs/community.md (hunk 7 above)
Comment (m-iwanicki) [reply docs/community.md]: fixed 38411e46ed59e835bd55a8351bc3755265a420b3

File: docs/steering-committee/.pages.yml (hunk 8 above)
Comment (m-iwanicki) [reply docs/steering-committee/.pages.yml]: fixed 38411e46ed59e835bd55a8351bc3755265a420b3

File: docs/CNAME (hunk 4 above)
Comment (m-iwanicki) [reply docs/CNAME]: Done in 667dfbb024eedcf1e1e5f44d44a33e18c48b9ab7

File: LICENSE.md (hunk 3 above)
Comment (m-iwanicki) [reply LICENSE.md]: Removed in 05c60db0af2f029bb3258c1ef1c28a64d35a6d19

File: LICENSE.md (hunk 2 above)
Comment (krystian-hebel) [reply LICENSE.md]: > But still we would need to use automatic ordering (each list point would be `1.` in raw text)

This isn't true:

```
1. a
   1. a
   2. a
   3. a
2. a
```

Renders as:

1. a
   1. a
   2. a
   3. a
2. a

Still, there is no way that I know of of using the roman numbering in source.

Perhaps renaming `LICENSE.md` to just `LICENSE` would solve some of the problems, but it may also create new ones. Let's leave it as is, it looks reasonably good in source form.

File: mkdocs.yml (hunk 11 above)
Comment (krystian-hebel) [reply mkdocs.yml:14]: Interesting, definitely not what I expected, and difficult for text editors without knowledge of YAML syntax. Since this seems to be the standard way of formatting YAML files, I'm marking it as resolved, even though I still don't like it :slightly_smiling_face:

File: docs/blueprints/Linux_Late_Launch.md [hunk 13]
```diff
@@ -1,39 +1,38 @@
...
-  * Process system state passed up from LZ
-    * e.g. memory map, etc.
-  * Setup Linux state for handover to mainline start up
-  * Jump into mainline start up
+* Sanitize any CPU state passed through late launch
+* Verify LZ signature with pubkey
+* Verify measurement from LZ signature against CRTM
+* (AMD) Create/Setup TPM Event log
+* Extend hash of pubkey into PCR 18, record in event log
+* This aligns to TXT's DA scheme
```
Comment (krystian-hebel) [inline docs/blueprints/Linux_Late_Launch.md]: This used to be on another indentation level.

File: docs/blueprints/Linux_Late_Launch.md [hunk 14]
```diff
@@ -1,39 +1,38 @@
...
+* Parse Linux header to detect relevant details about kernel
+* DEV/DMAR protect remainder of Linux kernel
+* Extend hash of Linux kernel into PCR 17, record in event log
+* Prepare and then handoff to Linux kernel proper
+* Need to support Legacy/CSM and UEFI environments
```
Comment (krystian-hebel) [inline docs/blueprints/Linux_Late_Launch.md]: Same here.

File: docs/blueprints/Linux_Late_Launch.md [hunk 15]
```diff
@@ -1,39 +1,38 @@
...
+* Add a new start/launch path to Linux kernel
+* Secure rendezvous APUs
+* Process system state passed up from LZ
+    - e.g. memory map, etc.
+* Setup Linux state for handover to mainline start up
+* Jump into mainline start up
```
Comment (krystian-hebel) [inline docs/blueprints/Linux_Late_Launch.md]: Same here.

File: docs/index.md [hunk 16]
```diff
@@ -4,6 +4,8 @@ hide:
   - toc
 ---
 
+# Home
```
Comment (krystian-hebel) [inline docs/index.md]: This was deliberately removed from the front page some time ago, it doesn't look good. Please add `<!-- markdownlint-disable MD041 -->` instead.

File: .markdownlint.yaml [hunk 17]
```diff
@@ -0,0 +1,196 @@
...
+
+# MD049/emphasis-style - Emphasis style should be consistent
+MD049:
+  # Emphasis style should be consistent
+  style: "underscore"
+
+# MD050/strong-style - Strong style should be consistent
+MD050:
+  # Strong style should be consistent
+  style: "asterisk"
```
Comment (krystian-hebel) [inline .markdownlint.yaml:196]: Can we add [MD053](https://github.com/DavidAnson/markdownlint/blob/main/doc/md053.md)? Would it catch errors like the one fixed by https://github.com/TrenchBoot/TrenchBoot.github.io/pull/46/commits/bc3ffcff801aaafd24f1da5cd457ac8338cce9df?

**Review by krystian-hebel — CHANGES_REQUESTED** [review]
Few more changes, and a rebase will be needed because another PR was merged.

File: .markdownlint.yaml (hunk 17 above)
Comment (m-iwanicki) [reply .markdownlint.yaml:196]: Yes:

```
markdownlint.............................................................Failed
- hook id: markdownlint
- exit code: 1

docs/blueprints/index.md:80:1 MD053/link-image-reference-definitions Link and image reference definitions should be needed [Duplicate link or image reference definition: "linux"] [Context: "[linux]: https://lore.kernel.o..."]
```

After changing:

```diff
-| Linux     | [In Progress][linux-up]         | N/A
+| Linux     | [In Progress][linux]         | N/A
 | Xen       | [Partially In Progress][xen-ap] | [In Progress][xen-qos]
 
 [grub-up]: https://lists.gnu.org/archive/html/grub-devel/2024-08/msg00088.html
 [grub-qos]: https://github.com/QubesOS/qubes-grub2/pull/13
-[linux-up]: https://lore.kernel.org/lkml/20240826223835.3928819-1-ross.philipson@oracle.com/
+[linux]: https://lore.kernel.org/lkml/20240826223835.3928819-1-ross.philipson@oracle.com/
```

Added in https://github.com/TrenchBoot/TrenchBoot.github.io/pull/37/commits/6959805f5ba51e7b7aeb1a826b4e58ff2fe821d1

File: docs/blueprints/Linux_Late_Launch.md (hunk 14 above)
Comment (m-iwanicki) [reply docs/blueprints/Linux_Late_Launch.md]: https://github.com/TrenchBoot/TrenchBoot.github.io/pull/37/commits/a31855e7af2ad0366cb920fbc483a3e8e3b92a83

File: docs/blueprints/Linux_Late_Launch.md (hunk 15 above)
Comment (m-iwanicki) [reply docs/blueprints/Linux_Late_Launch.md]: https://github.com/TrenchBoot/TrenchBoot.github.io/pull/37/commits/a31855e7af2ad0366cb920fbc483a3e8e3b92a83

File: docs/index.md (hunk 16 above)
Comment (m-iwanicki) [reply docs/index.md]: https://github.com/TrenchBoot/TrenchBoot.github.io/pull/37/commits/a31855e7af2ad0366cb920fbc483a3e8e3b92a83

File: docs/blueprints/Linux_Late_Launch.md (hunk 13 above)
Comment (m-iwanicki) [reply docs/blueprints/Linux_Late_Launch.md]: https://github.com/TrenchBoot/TrenchBoot.github.io/pull/37/commits/a31855e7af2ad0366cb920fbc483a3e8e3b92a83

File: docs/blueprints/Linux_Late_Launch.md (hunk 14 above)
Comment (m-iwanicki) [reply docs/blueprints/Linux_Late_Launch.md]: force-push: https://github.com/TrenchBoot/TrenchBoot.github.io/compare/a31855e7af2ad0366cb920fbc483a3e8e3b92a83..bd18e0db0237200a58a3d4bf0103b4ec187cfe7c

File: docs/blueprints/Linux_Late_Launch.md (hunk 13 above)
Comment (m-iwanicki) [reply docs/blueprints/Linux_Late_Launch.md]: force push: https://github.com/TrenchBoot/TrenchBoot.github.io/compare/a31855e7af2ad0366cb920fbc483a3e8e3b92a83..bd18e0db0237200a58a3d4bf0103b4ec187cfe7c

**Comment by m-iwanicki** [comment]
@krystian-hebel rebased and added fixes. Is it correct now? Indent was hard to see (2 spaces only and wasn't rendered on GitHub correctly)

**Review by krystian-hebel — APPROVED** [review]

**Review by krystian-hebel — APPROVED** [review]
